src/evaluate/module.py

```diff
@@ -706,3 +708,121 @@ def _enforce_nested_string_type(self, schema, obj):
+
+        self.force_prefix = force_prefix
+
+    def add(self, prediction=None, reference=None, **kwargs):
+        """Add one prediction and reference for each evaluation module's stack.
+
+        Args:
+            prediction (list/array/tensor, optional): Predictions.
+            reference (list/array/tensor, optional): References.
+        """
+        for evaluation_module in self.evaluation_modules:
+            batch = {"predictions": prediction, "references": reference, **kwargs}
+            batch = {input_name: batch[input_name] for input_name in evaluation_module._feature_names()}
+            evaluation_module.add(**batch)
+
+    def add_batch(self, predictions=None, references=None, **kwargs):
+        """Add a batch of predictions and references for each evaluation module's stack.
+
+        Args:
+            predictions (list/array/tensor, optional): Predictions.
+            references (list/array/tensor, optional): References.
+        """
+        for evaluation_module in self.evaluation_modules:
+            batch = {"predictions": predictions, "references": references, **kwargs}
+            batch = {input_name: batch[input_name] for input_name in evaluation_module._feature_names()}
+            evaluation_module.add_batch(**batch)
+
+    def compute(self, predictions=None, references=None, **kwargs):
+        """Compute each evaluation module.
+
+        Usage of positional arguments is not allowed to prevent mistakes.
+
+        Args:
+            predictions (list/array/tensor, optional): Predictions.
+            references (list/array/tensor, optional): References.
+            **kwargs (optional): Keyword arguments that will be forwarded to the evaluation module :meth:`_compute`
+                method (see details in the docstring).
+
+        Return:
+            dict or None
+
+            - Dictionary with the results if this evaluation module is run on the main process (``process_id == 0``).
+            - None if the evaluation module is not run on the main process (``process_id != 0``).
+        """
+        results = []
+
+        for evaluation_module in self.evaluation_modules:
+            batch = {"predictions": predictions, "references": references, **kwargs}
+            batch = {input_name: batch[input_name] for input_name in evaluation_module._feature_names()}
+            results.append(evaluation_module.compute(**batch))
+
+        return self._merge_results(results)
+
+    def _merge_results(self, results):
+        merged_results = {}
+        results_keys = list(itertools.chain.from_iterable([r.keys() for r in results]))
+        duplicate_keys = {item for item, count in collections.Counter(results_keys).items() if count > 1}
+
+        duplicate_names = [
+            item for item, count in collections.Counter(self.evaluation_module_names).items() if count > 1
+        ]
+        duplicate_counter = {name: 0 for name in duplicate_names}
```

Comment:
It's also possible to simply raise an error for now, if we need more time to think about how to handle this case.

As a user it's a bit unclear for me that it would automatically use prefixes when there are name conflicts, but no prefixes if there are no conflicts. This is the kind of black magic we try to avoid usually - though I don't have a strong opinion on this very specific case, since it would be easy to explain in the docs. What do you think ?

an alternative could be to ask users to name the metrics:
```python
combine({
    "metric_1": metric_1,
    "metric_2": metric_2
})
```

Reply:
What do you think about adding a warning when a prefixes are added:
```py
f"The key '{key}' occurs in several evaluation results. Prefixing with the key with '{prefix}'."
```

The user can already use a `dict` in the current state of the PR (see last example in initial comment). What do you think?

Reply:
A warning sounds good thanks !